valhalla/midgard/point_tile_index.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <unordered_set>
+
+#include "midgard/linesegment2.h"
+#include "midgard/pointll.h"
+#include "midgard/tiles.h"
+
+namespace valhalla {
+
+namespace midgard {
+
+/**
+ * Provided a search width (tile_width_degrees) and a bunch of points,
+ * this class will bin the points into a gridded/tiled space at that width.
+ *
+ * Once all the points are binned we can quickly solve problems like "what
+ * points are near me" - in effectively O(1) time (the speed of a hash
+ * lookup).
+ *
+ * Of important note is the fact that to determine the "points within an
+ * exact distance of a given point" you must 1) use the get_points_near*()
+ * methods to round-up nearby points, then 2) compute the distance those
+ * nearby points using a more accurate distance computation of your choosing.
+ */
+class PointTileIndex {
+  // This guy can tell us which tile a points belongs to.
+  std::unique_ptr<Tiles<PointLL>> tiles;
```

Comment:
Switched to using the `Tiles` class for creating a gridded space over the given points-of-interest.  Works well and performs equal to my previous solution.